core/arch/aarch64/proc.c

```diff
@@ -99,12 +101,42 @@ get_processor_specific_info(void)
     cpu_info.features.flags_aa64dfr0 = isa_features[AA64DFR0];
     cpu_info.features.flags_aa64zfr0 = isa_features[AA64ZFR0];
 
-#        if !defined(DR_HOST_NOT_TARGET) && defined(SVE)
-    /* TODO i#3044: Vector length will be set by reading value from h/w. */
-    CLIENT_ASSERT(false, "TODO i#3044: SVE requires initialisation of vector length!");
-#        elif !defined(STANDALONE_DECODER) || defined(DR_HOST_NOT_TARGET)
-    /* Set SVE vector length for unit tests. */
-    dr_set_sve_vl(256);
+    /* The SVE vector length is set to:
+     * - A value read from the host hardware.
+     * or:
+     * - 32 bytes, 256 bits.
+     * Which of the above depends on:
+     * - SVE or non-SVE AArch64 or x86 host h/w.
+     * and:
+     * - Release or development test build.
+     */
+#        if !defined(DR_HOST_NOT_TARGET)
+    if (proc_has_feature(FEATURE_SVE)) {
+#            if !defined(BUILD_TESTS)
```

Comment:
I don't think we want this #if.  We want DR's behavior to be the same regardless of whether tests were built or not.  With this as-is here aren't we never testing this RDVL in any test suite?

Reply:
The if `!defined(BUILD_TESTS)` is there because when running codec unit tests on SVE hardware we have to use a fixed/constant VL rather than whatever VL the hardware supports.

The `RDVL` and hence the VL implemented by SVE hardware is not tested upstream yet because we don't have SVE hardware on which to run precommit testing. We have tested this branch internally on SVE hardware on which `RDVL` is executed and all code enabled by `proc_has_feature(FEATURE_SVE)`.